### ISSUE TITLE
Videos won't continue, always start from beginning

### ISSUE THREAD

**Issue by schnills** [FriedChickenButt/youtube-webos#31]
Hey,
first of all: thanks for using YouTube usable at all again. It became unbearable and your app has even better features than premium (sponsor skipping etc). 
Only one issue that's a bit annoying: It doesn't save where I stopped a video. It always displays a seen video with the red bar at the beginning at always starts from there. So I can't really continue where I stopped but have to remember where that was. 
I mean, this is a relatively small trade off for the otherwise great features you provide. I'd always choose this over ads every three minutes. 
Still maybe you can improve this in the future :) 

Another thing you might find interesting is how I keep developer mode active without root. I've described it here:
https://forum.xda-developers.com/t/developer-mode-extend-session-time.4012841/

Thanks for your great work.

**Comment by Informatic** [comment]
Have you tried disabling adblocking and see if that works again? We are definitely overeager with requests we block (one such is `ptracking`) - after removing that / disabling adblocking that request (and playback progress requests) are properly sent, *but* I wasn't able to make the player actually start from the last played timestamp (but "normal" YouTube running on a separate computer with the same Google account signed in did in fact start from a proper timestamp)

Which webOS version are you on? 4.x+? The way we launch youtube is similar to what was done on 3.x and older. (running in WebAppManager, instead of [Cobalt](https://cobalt.foo/)) Maybe launching from a proper timestamp is just broken on older browsers.

Reference webosbrew/youtube-webos#4 - same issue.

**Comment by schnills** [comment]
Thanks for the quick response, man! 
Disabling adblocking indeed made it working again. It continued at the corresponding timestamp. After enabling adblocking again, no new timestamp was saved - the old one from having adblocking disabled is still active. 

I'm on WebOS 4.30.40. 
In the reference you posted I saw a test ipk. I'll probably give it a try tomorrow. 

Btw, I just upgraded from youtube adfree v4_0.1.0 (or .1.1) to v4_0.2.0 and realized the settings have been reduced. I kept them all on anyway, but any specific reason to reduce them only to ad and sponsor blocking? 
I could imagine some people would like to have more options about intro/outro skipping and such.

**Comment by Informatic** [comment]
Not sure which version you are talking about. `0.2.0` was never released. Settings menu should look exactly like this: https://github.com/webosbrew/youtube-webos#readme
If you actually downloaded that test `.ipk` and it's missing, then that's definitely a bug (I didn't notice on my TV during testinG)

**Comment by schnills** [comment]
Uhm no, I actually downloaded it from your repository 
![image](https://user-images.githubusercontent.com/65947552/146831029-375c0813-b97d-466a-b0e6-088fbfa2635e.png)

**Comment by schnills** [comment]
I just stumbled across it yesterday and thought it's new public version

**Comment by Informatic** [comment]
Nope, that's a gross misunderstanding, sorry. I was just testing out github releases on my fork a month or so ago. :D
Test release I mentioned can be downloaded here: https://github.com/webosbrew/youtube-webos/actions/runs/1603263259

**Comment by schnills** [comment]
Ah okay :D 
Sorry for keeping you waiting, but I could test it now and it seems to work just fine, which is great. 
Thanks so much :)

**Comment by Informatic** [comment]
Great. A version with this (and a couple other fixes) has been released in webosbrew fork/repo.

**Comment by dryna2017** [comment]
Сейчас новое обновления 0.2.1 и видео не сохраняет с того места где я остановил его, начинается с начало это начинает раздражать постоянно перематывать, вы можете исправить эту ошибку

**Comment by dryna2017** [comment]
Now the new update is 0.2.1 and the video is not saved from the place where I stopped it, it starts from the beginning it starts annoying to constantly rewind, you can fix this error.

**Comment by dryna2017** [comment]
Hello, I really like this YouTube, there is only one problem that I would like you to fix, the place where I stopped the video is not saved, and when I log in, the video starts from the beginning, there is one account on all devices and everything works and is saved on them, I hope for you that you will check everything correctly and fix this error. Thank you if you read and help.

**Comment by strdik** [comment]
Мне нравится этот ютуб но невозможно смотреть когда не сохраняет видео, пожалуйста если прочитаете этот комментарий проверьте все как следует исправте ошибки и сделайте что бы сохранялось видео на том месте где остановил сейчас стоит версия 0.2.1  я надеюсь на вас и с наступающим.

**Comment by strdik** [comment]
I like this YouTube, but it is impossible to watch when it does not save the video, please, if you read this comment, check everything properly, fix the errors and make sure that the video is saved at the place where it stopped now, version 0.2.1 is standing, I hope for you and happy New Year.
